Ensure graceful handling of invalid / missing background image in theme

Splitting out as an issue from #144:

example of what a missing file does:
```
Error: Cannot find module './bg-045.svg'.
.svg:54
The above error occurred in the <ThemeBackgroundPicker> component:
    in ThemeBackgroundPicker (created by OnClickOutside(ThemeBackgroundPicker))
    in OnClickOutside(ThemeBackgroundPicker) (created by AppComponent)
    in div (created by AppComponent)
    in section (created by BrowserPreview)
    in div (created by BrowserPreview)
    in BrowserPreview (created by AppComponent)
    in main (created by AppComponent)
    in div (created by AppComponent)
    in AppComponent (created by Connect(AppComponent))
    in Connect(AppComponent)
    in Provider

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
react-dom.development.js:9747
Error: Cannot find module './bg-045.svg'.
.svg:54
```

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "themer",
   "description": "Theming demo for Firefox Quantum and beyond.",
-  "version": "0.0.30",
+  "version": "0.0.31",
   "author": "Les Orchard <me@lmorchard.com>",
   "contributors": [
     "John Gruen <john.gruen@gmail.com>"

--- a/src/extension/background.js
+++ b/src/extension/background.js
@@ -1,5 +1,10 @@
 import { makeLog } from "../lib/utils";
-import { normalizeTheme, colorToCSS, bgImages } from "../lib/themes";
+import {
+  normalizeTheme,
+  normalizeThemeBackground,
+  colorToCSS,
+  bgImages
+} from "../lib/themes";
 
 // Blank 1x1 PNG from http://png-pixel.com/
 const BLANK_IMAGE =
@@ -11,15 +16,14 @@ const siteUrl = process.env.SITE_URL;
 
 const init = () => {
   browser.browserAction.onClicked.addListener(() => {
-    browser.tabs.query({ currentWindow: true })
-      .then(tabs => {
-        const themerTab = tabs.find(tab => tab.url.includes(siteUrl));
-        if (themerTab) {
-          browser.tabs.update(themerTab.id, { active: true });
-        } else {
-          browser.tabs.create({ url: siteUrl });
-        }
-      });
+    browser.tabs.query({ currentWindow: true }).then(tabs => {
+      const themerTab = tabs.find(tab => tab.url.includes(siteUrl));
+      if (themerTab) {
+        browser.tabs.update(themerTab.id, { active: true });
+      } else {
+        browser.tabs.create({ url: siteUrl });
+      }
+    });
   });
   browser.runtime.onConnect.addListener(port => {
     port.onMessage.addListener(messageListener(port));
@@ -62,9 +66,7 @@ const applyTheme = ({ theme }) => {
   }
 
   const background = theme.images.additional_backgrounds[0];
-  const backgroundImage = bgImages.keys().includes(background)
-    ? bgImages(background)
-    : "images/patterns/bg-000.svg";
+  const backgroundImage = bgImages(normalizeThemeBackground(background));
 
   const newTheme = {
     images: {

--- a/src/lib/store.js
+++ b/src/lib/store.js
@@ -69,7 +69,10 @@ export const reducers = {
       SET_SAVED_THEMES: (state, { payload: { savedThemes } }) => ({
         ...state,
         savedThemes: Object.entries(savedThemes).reduce(
-          (acc, [key, theme]) => ({ ...acc, [key]: { theme: normalizeTheme(theme) } }),
+          (acc, [key, savedTheme]) => ({
+            ...acc,
+            [key]: { ...savedTheme, theme: normalizeTheme(savedTheme.theme) }
+          }),
           {}
         )
       }),

--- a/src/lib/themes.js
+++ b/src/lib/themes.js
@@ -4,6 +4,12 @@ const presetThemesContext = require.context(
   /.*\.json/
 );
 
+export const bgImages = require.context(
+  "../images/patterns/",
+  false,
+  /bg-.*\.svg/
+);
+
 const defaultTheme = presetThemesContext("./default.json");
 
 export const colorToCSS = color => {
@@ -12,6 +18,9 @@ export const colorToCSS = color => {
     ? `hsl(${h}, ${s}%, ${l}%)`
     : `hsla(${h}, ${s}%, ${l}%, ${a * 0.01})`;
 };
+
+export const normalizeThemeBackground = background =>
+  bgImages.keys().includes(background) ? background : "";
 
 // Utility to ensure normal & consistent colors
 export const normalizeThemeColor = (data, defaultColor) => {
@@ -38,20 +47,20 @@ export const normalizeTheme = (data = {}) => {
   const theme = {
     colors: normalizeThemeColors(data.colors, defaultTheme.colors),
     images: {
-      additional_backgrounds: []
+      additional_backgrounds: [ "" ]
     }
   };
   const images = data.images ? data.images : {};
   if (images.headerURL) {
-    theme.images.additional_backgrounds = [ images.headerURL ];
+    theme.images.additional_backgrounds = [ normalizeThemeBackground(images.headerURL) ];
   }
   if (images.additional_backgrounds) {
-    theme.images.additional_backgrounds = images.additional_backgrounds;
+    theme.images.additional_backgrounds = [
+      normalizeThemeBackground(images.additional_backgrounds[0])
+    ];
   }
   return theme;
 };
-
-export const bgImages = require.context("../images/patterns/", false, /bg-.*\.svg/);
 
 export const presetThemes = presetThemesContext
   .keys()

--- a/src/web/lib/storage.js
+++ b/src/web/lib/storage.js
@@ -1,5 +1,6 @@
 import { actions } from "../../lib/store";
 import { makeLog } from "../../lib/utils";
+import { normalizeTheme } from "../../lib/themes";
 
 const log = makeLog("web.storage");
 
@@ -37,7 +38,9 @@ function deleteTheme(key) {
 
 function getTheme(key) {
   try {
-    return JSON.parse(localStorage.getItem(themeStorageKey(key)));
+    const value = JSON.parse(localStorage.getItem(themeStorageKey(key)));
+    value.theme = normalizeTheme(value.theme);
+    return value;
   } catch (e) {
     return null;
   }

